src/Microsoft.OData.Core/JsonLight/ODataJsonLightWriter.cs

```diff
@@ -990,7 +990,10 @@ protected override void EndNestedResourceInfoWithContent(ODataNestedResourceInfo
             // resource set afterwards, we have to now close the array of links.
             if (navigationLinkScope.EntityReferenceLinkWritten && !navigationLinkScope.ResourceSetWritten && nestedResourceInfo.IsCollection.Value)
             {
-                this.jsonWriter.EndArrayScope();
+                if (!this.jsonLightOutputContext.WritingResponse)

```

Comment:
why make this a separate if, and not part of the above condition (perhaps with line breaks inserted for readability)?

Reply:
Resolved.

---
In reply to: [289888907](https://github.com/OData/odata.net/pull/1464#discussion_r289888907) [](ancestors = 289888907,285800014)